### PR TITLE
Update Gradle plugin

### PR DESCRIPTION
**Pull request by nhachicha**
- Update Gradle plugin
- Update Kotlin runtime 

This should allow you to use AS 3.1 Canary 5

### CODE REVIEW

File: realm/build.gradle [hunk 1]
```diff
@@ -3,7 +3,7 @@ project.ext.compileSdkVersion = 26
 project.ext.buildToolsVersion = '26.0.2'
 
 buildscript {
-    ext.kotlin_version = '1.1.51'
+    ext.kotlin_version = '1.2.0'
```
Comment (zaki50) [inline realm/build.gradle]: `1.2.10` has been released

File: realm/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,6 @@
+#Wed Dec 13 12:08:46 GMT 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.3.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions-snapshots/gradle-4.4-20171031235950+0000-all.zip
```
Comment (zaki50) [inline realm/gradle/wrapper/gradle-wrapper.properties]: The official version of Gradle 4.4 is already released.

And we should not edit `gradle-wrapper.properties` directly. See https://github.com/realm/realm-java/blob/master/README.md#upgrading-gradle-wrappers

**Comment by zaki50** [comment]
Shouldn't this PR target `releases` branch?

File: library-benchmarks/build.gradle [hunk 3]
```diff
@@ -28,7 +28,7 @@ apply plugin: 'realm-android'
 
 android {
     compileSdkVersion 27
-    buildToolsVersion "27.0.0"
+    buildToolsVersion "27.0.1"
```
Comment (zaki50) [inline library-benchmarks/build.gradle]: latest is `27.0.2`

**Comment by zaki50** [comment]
@nhachicha `README.md` also should be updated.

`Download & install the Android SDK Build-Tools 27.0.1`
↓
`Download & install the Android SDK Build-Tools 27.0.2`

**Comment by Sirrah** [comment]
The Kotlin stdlib was renamed during the 1.2.x release. See [here](https://blog.jetbrains.com/kotlin/2017/09/kotlin-1-2-beta-is-out/). The old name `kotlin-stdlib-jre7` was changed to `kotlin-stdlib-jdk7`.
This library is referenced in realm-library and kotlin-extensions.

Sorry I didn't include a patch, but there are already multiple PRs open for updating Kotlin.

**Pull request merged**